website/content/docs/integrations/consul/index.mdx

```diff
@@ -0,0 +1,182 @@
+
+## Service Mesh
+
+Consul service mesh provides service-to-service connection authorization and
+encryption using mutual Transport Layer Security (TLS). Nomad can automatically
+provision the components necessary to securely connect your tasks to Consul's
+service mesh.
+
+Refer to the [Consul Service Mesh integration page][int_consul_mesh] for more
+information.
+
+## Dynamic Configuration
+
+Nomad's job specification includes a [`template`][] block that uses a Consul
+ecosystem tool called [Consul Template][]. This mechanism creates a convenient
+way to ship configuration files that are populated from environment variables,
+Consul data, Vault secrets, or just general configurations within a Nomad task.
+
+For more information on Nomad's template block and how it leverages Consul
+Template, please see the [`template` job specification documentation][].
+
+## Consul ACL
+
+The Consul ACL system protects the cluster from unauthorized access. When
+enabled, both Consul and Nomad must be properly configured in order for their
+integrations to work.
+
+Refer to the [Consul ACL integration page][int_consul_acl] for more
+information.
+
+## Consul Namespaces
+
+Nomad provides integration with [Consul Namespaces][consul_namespaces] for
+service registrations specified in `service` blocks and Consul KV reads in
+`template` blocks.
+
+By default, Nomad will not specify a Consul namespace on service registrations
+or KV store reads, which Consul then implicitly resolves to the `"default"`
+namespace.  This default namespace behavior can be modified by setting the
+[`namespace`][consul_agent_namespace] field in the Nomad agent Consul
+configuration block.
+
+For more control over Consul namespaces, Nomad Enterprise supports configuring
+the Consul [namespace][consul_jobspec_namespace] at the group or task level in
+the Nomad job spec as well as the [`-consul-namespace`][consul_run_namespace]
+command line argument for `job run`.
+
+The Consul namespace used for a set of group or task service registrations
+within a group, as well as `template` KV store access is determined from the
+following hierarchy from lowest to highest precedence:
+
+* Consul default: If no Consul namespace options are configured, Consul will
+  automatically make use of the `"default"` namespace.
+
+* agent configuration: If the [`namespace`][consul_agent_namespace] Nomad agent
+  Consul configuration parameter is set, this namespace will be used instead of
+  the Consul default.
+
+* job run command: <EnterpriseAlert inline/> If the
+  [`-consul-namespace`][consul_run_namespace] command line argument is specified
+  on job submission, this namespace will take precedence over the namespace set
+  in Nomad agent configuration.
+
+* group and task configuration: <EnterpriseAlert inline/> If the group level or
+  task level Consul [namespace field][consul_jobspec_namespace] is configured,
+  this namespace will take precedence over all other options.
```

Comment:
Good point. I actually didn't even read this part, just copy-and-pasted from the original source 😅 

Re-order and re-worded so it sounds less if-else conditional. Also added the `CONSUL_NAMESPACE` env var since it's not mentioned in the `-consul-namespace` flag docs.

<img width="714" alt="image" src="https://github.com/hashicorp/nomad/assets/775380/3f7399a9-be9e-4fc9-9b22-4e3da9c5ca4d">

Reply:
That works!